zap/src/main/java/org/zaproxy/zap/control/AddOn.java

```diff
@@ -1372,6 +1372,18 @@ private static void calculateExtensionRunRequirements(
             calculateRunRequirementsImpl(
                     availableAddOns, extensionRequirements, addOn, addOnDep, checkLibs);
         }
+
+        for (var dependency : extension.getExtensionDependencies()) {
+            String classname = dependency.getClassname();
+            if (classname == null) {
+                continue;
+            }
+            if (addOn.getExtensionsWithDeps().contains(classname)) {
```

Comment:
It should also check that it can be loaded (i.e. the transitive dependencies are all met).